iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/ValidationUtil.java

```diff
@@ -52,18 +52,31 @@ static void checkActivePowerLimits(Validable validable, double minP, double maxP
         }
     }
 
-    static void checkVoltageControl(Validable validable, Boolean voltageRegulatorOn, double voltageSetpoint, double reactivePowerSetpoint) {
+    static void checkTargetDeadband(Validable validable, String validableType, double targetDeadband) {
+        if (Double.isNaN(targetDeadband)) {
+            throw new ValidationException(validable, "Undefined value for target deadband of " + validableType);
+        }
+        if (targetDeadband < 0) {
+            throw new ValidationException(validable, "Unexpected value for target deadband of " + validableType + ": " + targetDeadband + " < 0");
+        }
+    }
+
+    static boolean checkVoltageControl(Validable validable, Boolean voltageRegulatorOn, double voltageSetpoint) {
         if (voltageRegulatorOn == null) {
             throw new ValidationException(validable, "voltage regulator status is not set");
         }
         if (voltageRegulatorOn) {
             if (Double.isNaN(voltageSetpoint) || voltageSetpoint <= 0) {
                 throw createInvalidValueException(validable, voltageSetpoint, "voltage setpoint", "voltage regulator is on");
             }
-        } else {
-            if (Double.isNaN(reactivePowerSetpoint)) {
-                throw createInvalidValueException(validable, reactivePowerSetpoint, "reactive power setpoint", "voltage regulator is off");
-            }
+            return false;
+        }
+        return true;
+    }
+
+    static void checkVoltageControl(Validable validable, Boolean voltageRegulatorOn, double voltageSetpoint, double reactivePowerSetpoint) {
+        if (checkVoltageControl(validable, voltageRegulatorOn, voltageSetpoint) && Double.isNaN(reactivePowerSetpoint)) {
+            throw createInvalidValueException(validable, reactivePowerSetpoint, "reactive power setpoint", "voltage regulator is off");
         }
```

Comment:
I'm not sure to understand why these method is split in two parts...

Reply:
I am only using `checkVoltageControl(Validable, Boolean, double)` for shunts since they don't have `reactivePowerSetpoint` and it was to avoid duplication. But for two lines, I could do a specific method I guess if this is too confusing.